### PR TITLE
Added Framework contract schema

### PR DESCRIPTION
**Pull request by Maxy673**
- Added schema to framework contract and related table to project ID

### CODE REVIEW

**Review by IvanRomero03 — CHANGES_REQUESTED** [review]
please document the pr with the correct title name and correct description.

File: src/app/projects/[projectId]/Details.tsx [hunk 1]
```diff
@@ -36,6 +36,7 @@ import {
 } from "techme/util/Readables";
 import { type UserRole, readableRole } from "techme/util/UserRole";
 import { api } from "techme/trpc/react";
+import React from "react";
```
Comment (IvanRomero03) [inline src/app/projects/[projectId]/Details.tsx]:
```suggestion
```
unused import

File: src/app/projects/[projectId]/Analysis.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import React from 'react'
+
+const Analysis = () => {
+  return (
+    <div>Analysis</div>
+  )
+}
+
+export default Analysis
```
Comment (IvanRomero03) [inline src/app/projects/[projectId]/Analysis.tsx]: unused/unfinished component

File: src/app/projects/[projectId]/Estimations.tsx [hunk 3]
```diff
@@ -1,5 +1,7 @@
 "use client";
 
+import React from "react";
+
```
Comment (IvanRomero03) [inline src/app/projects/[projectId]/Estimations.tsx]: unused import

File: src/app/projects/[projectId]/page.tsx [hunk 4]
```diff
@@ -1,12 +1,15 @@
 "use client";
 
+
+
```
Comment (IvanRomero03) [inline src/app/projects/[projectId]/page.tsx]: .

File: src/app/projects/[projectId]/page.tsx [hunk 5]
```diff
@@ -1,12 +1,15 @@
...
 import { cn } from "lib/utils";
 import { useState } from "react";
 import { api } from "techme/trpc/react";
 import Details from "./Details";
 import Estimations from "./Estimations";
 import Requirements from "./Requirements";
 import Summary from "./Summary";
+import React from "react";
```
Comment (IvanRomero03) [inline src/app/projects/[projectId]/page.tsx]: same here and below

File: src/server/api/root.ts [hunk 6]
```diff
@@ -16,11 +17,15 @@ export const appRouter = createTRPCRouter({
...
   prjectTasks: projectsRouterTasks,
+  frameworkContracts: frameworkContractsRouter,
 });
 
 // export type definition of API
 export type AppRouter = typeof appRouter;
 
+
+
+
```
Comment (IvanRomero03) [inline src/server/api/root.ts]: please dont leave these blank spaces

File: src/server/api/routers/frameworkContractsRouter.ts [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+import { frameworkContracts, frameworkContractPerProject } from "techme/server/db/schema"; 
+
+
+export const frameworkContractsRouter = createTRPCRouter({
+  getAllContracts: protectedProcedure
+    .input(z.object({ projectId: z.number() }))
+    .query(async ({ ctx, input }) => {
+      
+      return await ctx.db.select().from(frameworkContracts)
+        .where(eq(frameworkContractPerProject.projectId, input.projectId)) // Obtain contracts filter
```
Comment (IvanRomero03) [inline src/server/api/routers/frameworkContractsRouter.ts]:
```suggestion
        .where(eq(frameworkContractPerProject.projectId, input.projectId)) 
```

File: src/server/api/routers/frameworkContractsRouter.ts [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    }),
+
+
+    // Should i add other mutations or functions ivan?? 
+
+  
+});
+
+
+
```
Comment (IvanRomero03) [inline src/server/api/routers/frameworkContractsRouter.ts]: blank spaces

**Review by IvanRomero03 — APPROVED** [review]

**Review by IvanRomero03 — CHANGES_REQUESTED** [review]
please review the comments

File: src/app/projects/[projectId]/Details.tsx [hunk 9]
```diff
@@ -37,6 +37,7 @@ import {
 import { type UserRole, readableRole } from "techme/util/UserRole";
 import { api } from "techme/trpc/react";
 
+
```
Comment (IvanRomero03) [inline src/app/projects/[projectId]/Details.tsx]:
```suggestion
```

File: src/app/projects/[projectId]/Estimations.tsx [hunk 10]
```diff
@@ -1,5 +1,7 @@
 "use client";
 
+
+
```
Comment (IvanRomero03) [inline src/app/projects/[projectId]/Estimations.tsx]:
```suggestion
```

File: src/app/projects/[projectId]/page.tsx [hunk 11]
```diff
@@ -1,5 +1,7 @@
 "use client";
 
+
+
```
Comment (Maxy673) [inline src/app/projects/[projectId]/page.tsx]:
```suggestion
```

File: src/app/projects/[projectId]/page.tsx [hunk 12]
```diff
@@ -8,6 +8,7 @@ import Estimations from "./Estimations";
 import Requirements from "./Requirements";
 import Summary from "./Summary";
 
+
```
Comment (Maxy673) [inline src/app/projects/[projectId]/page.tsx]:
```suggestion
```

**Review by IvanRomero03 — APPROVED** [review]
lgtm

**Pull request merged**